src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -2881,6 +2888,176 @@ public ConstructorEdit(INamedTypeSymbol oldType)
+
+            FindChangedAttributes(oldAttributes, newAttributes, changedAttributes);
+            if (oldAttributes.HasValue)
+            {
+                FindChangedAttributes(newAttributes, oldAttributes.Value, changedAttributes);
+            }
+
+            needsEdit = false;
+            if (changedAttributes.Count == 0)
+            {
+                return;
+            }
+
+            // We need diagnostics reported if the runtime doesn't support changing attributes,
+            // but even if it does, only attributes stored in the CustomAttributes table are editable
+            if (!capabilities.HasFlag(EditAndContinueCapabilities.UpdateCustomAttributes) ||
+                changedAttributes.Any(IsNonCustomAttribute))
+            {
+                var newNode = FindSyntaxNode(newSymbol);
+                diagnostics.Add(new RudeEditDiagnostic(RudeEditKind.ChangingAttributesNotSupportedByRuntime, GetDiagnosticSpan(newNode, EditKind.Update), newNode, new[] { GetDisplayName(newNode, EditKind.Update) }));
+            }
+            else
+            {
+                needsEdit = true;
+            }
+
+            return;
+
+            static void FindChangedAttributes(ImmutableArray<AttributeData>? oldAttributes, ImmutableArray<AttributeData> newAttributes, ArrayBuilder<AttributeData> changedAttributes)
+            {
+                for (var i = 0; i < newAttributes.Length; i++)
+                {
+                    var newAttribute = newAttributes[i];
+                    var oldAttribute = FindMatch(newAttribute, oldAttributes);
+
+                    if (oldAttribute is null)
+                    {
+                        changedAttributes.Add(newAttribute);
+                    }
+                }
+            }
+
+            static AttributeData? FindMatch(AttributeData attribute, ImmutableArray<AttributeData>? oldAttributes)
+            {
+                if (!oldAttributes.HasValue)
+                {
+                    return null;
+                }
+
+                foreach (var match in oldAttributes.Value)
+                {
+                    if (SymbolEquivalenceComparer.Instance.Equals(match.AttributeClass, attribute.AttributeClass))
+                    {
+                        if (SymbolEquivalenceComparer.Instance.Equals(match.AttributeConstructor, attribute.AttributeConstructor) &&
+                            match.ConstructorArguments.SequenceEqual(attribute.ConstructorArguments) &&
+                            match.NamedArguments.SequenceEqual(attribute.NamedArguments))
+                        {
+                            return match;
+                        }
+                    }
+                }
+                return null;
+            }
+
+            static SyntaxNode FindSyntaxNode(ISymbol symbol)
+            {
+                // In VB parameters of delegates don't have declaring syntax references so we have to go all the way up to the delegate
```

Comment:
> // In VB parameters of delegates don't have declaring syntax references so we have to go all the way up to the delegate [](http://example.com/codeflow?start=15&length=120)

Isn't this a bug?

Reply:
Yes: https://github.com/dotnet/roslyn/issues/53337